### PR TITLE
Document image purpose

### LINKED ISSUE #235: Document the purpose of the images
Create documentation about the purpose of the images.

### PR DESCRIPTION
**Pull request by RDIL**
## PR Summary
Fixes #235 

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
  - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `Dockerfile`, `.sh`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
  - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **Adds a new image or tag**
  - [x] Not Applicable
- **OR**
  - [ ] Update [README.powershellcommunity.md](https://github.com/PowerShell/PowerShell-Docker/blob/master/assets/README.powershellcommunity.md)
  - [ ] Update [vsts-ci.yml](https://github.com/PowerShell/PowerShell-Docker/blob/master/vsts-ci.yml)

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# PowerShell-Docker
+
+Hello!
+This is the main documentation page for PowerShell-Docker, so here you can find some helpful details for different questions.
+
+## Image Purpose
+
+These images are built so that PowerShell users can run the program in a containerized environment - see [this article](https://opensource.com/resources/what-docker) for what Docker is, and some basic pros and cons.
```
Comment (TravisEz13) [inline docs/index.md]: 1. The main image is intended to be minimal.  In other words, if you bring a module, you may need to add libraries to the image.

2. The test-dep sub images are intended to allow running tests in Azure Dev Ops

**Comment by TravisEz13** [comment]
Thanks for starting this.

**Comment by TravisEz13** [comment]
Thanks, I think you stated this better than I did.

**Comment by RDIL** [comment]
No worries :)

**Comment by TravisEz13** [comment]
And I'm glad you are still working in GitHub!

**Comment by RDIL** [comment]
Screenshot 2019-10-10 at 5 08 49 PM
I don't think I'm leaving anytime soon haha

**Comment by TravisEz13** [comment]
I thought I saw a Tweet...

**Comment by RDIL** [comment]
Well I am trying to move primary development over to GitLab, but a ton of projects are still here so I will be going between the two

**Comment by RDIL** [comment]
@PoshChan please rebuild CI

**Comment by PoshChan** [comment]
@RDIL, successfully started rebuild of `docker-PR`

**Comment by RDIL** [comment]
oh yes I'm going to have a lot of fun with this :>

**Comment by TravisEz13** [comment]
@RDIL Please don't abuse it or I'll have to remove you.  Please use retry instead of rebuild when possible, especially in this repo.

**Comment by TravisEz13** [comment]
Rebuild = redo everything
Retry = retry what has failed

**Comment by RDIL** [comment]
Okay, I was just joking

**Comment by TravisEz13** [comment]
Oh, I took it that way.

**Pull request merged**